### PR TITLE
Use zyte_spider_templates.Addon and clean the current settings

### PR DESCRIPTION
**Pull request by PyExplorer**

### CODE REVIEW

File: zyte_spider_templates_project/settings.py [hunk 1]
```diff
@@ -6,28 +6,28 @@
...
+    "zyte_spider_templates.Addon": 100,
     "scrapy_zyte_api.Addon": 500,
     "duplicate_url_discarder.Addon": 600,
 }
-SCHEDULER_DISK_QUEUE = "scrapy.squeues.PickleFifoDiskQueue"
-SCHEDULER_MEMORY_QUEUE = "scrapy.squeues.FifoMemoryQueue"
+
+DOWNLOADER_MIDDLEWARES = {
+    "scrapy_poet.InjectionMiddleware": 543,
+    "scrapy_zyte_api.ScrapyZyteAPIDownloaderMiddleware": 1000,
```
Comment (kmike) [inline zyte_spider_templates_project/settings.py]: Do we scrapy_zyte_api middlewares here, as we have scrapy_zyte_api.Addon enabled?

Comment (PyExplorer) [reply zyte_spider_templates_project/settings.py]: In addon we don't have `ScrapyZyteAPIDownloaderMiddleware` and I left it here.
@Gallaecio as you initially prepared the Addon - do you think this middleware not needed here?

Comment (Gallaecio) [reply zyte_spider_templates_project/settings.py]: `zyte_spider_templates.Addon` does not set `ScrapyZyteAPIDownloaderMiddleware`, but `scrapy_zyte_api.Addon` does, so we don’t need it here. Same for `scrapy_zyte_api.ScrapyZyteAPISpiderMiddleware` below.

File: zyte_spider_templates_project/settings.py [hunk 2]
```diff
@@ -6,28 +6,28 @@
 BOT_NAME = "zyte_spider_templates_project"
 
 SPIDER_MODULES = [
-    "zyte_spider_templates.spiders",
```
Comment (Gallaecio) [inline zyte_spider_templates_project/settings.py:9]: I wonder if we can really do this, thinking about the recent https://github.com/scrapy/scrapy/issues/6544

File: zyte_spider_templates_project/settings.py [hunk 3]
```diff
@@ -6,28 +6,28 @@
...
     "zyte_spider_templates_project.spiders",
 ]
 NEWSPIDER_MODULE = "zyte_spider_templates_project.spiders"
 
-CLOSESPIDER_TIMEOUT_NO_ITEM = 900
 ADDONS = {
+    "zyte_spider_templates.Addon": 100,
```
Comment (Gallaecio) [inline zyte_spider_templates_project/settings.py]: I wonder if we should put it after the other add-ons, as this add-on probably expects scrapy-zyte-api to be configured already, which the add-on below does. Maybe that’s why you ended up adding the download handler manually?

File: zyte_spider_templates_project/settings.py (hunk 2 above)
Comment (PyExplorer) [reply zyte_spider_templates_project/settings.py:9]: Fixed!

File: zyte_spider_templates_project/settings.py (hunk 3 above)
Comment (PyExplorer) [reply zyte_spider_templates_project/settings.py]: Fixed!

File: zyte_spider_templates_project/settings.py (hunk 1 above)
Comment (PyExplorer) [reply zyte_spider_templates_project/settings.py]: Fixed!

File: zyte_spider_templates_project/settings.py (hunk 2 above)
Comment (Gallaecio) [reply zyte_spider_templates_project/settings.py:9]: You have removed the entire setting. My point is that we probably need to keep it as it is, no changes.

Comment (PyExplorer) [reply zyte_spider_templates_project/settings.py:9]: Do you mean keep
```
SPIDER_MODULES = [
    "zyte_spider_templates.spiders"
]?
```
I as got from the https://github.com/scrapy/scrapy/issues/6544 it won't work this way.

Comment (Gallaecio) [reply zyte_spider_templates_project/settings.py:9]: I mean keep:

```
SPIDER_MODULES = [
    "zyte_spider_templates.spiders",
    "zyte_spider_templates_project.spiders",
]
```

I assume this is the only way you can have both spiders from zyte-spider-templates and custom spiders from your project work both locally (e.g. get listed by `scrapy list`) and in Scrapy Cloud. But better test it out to confirm.

Comment (PyExplorer) [reply zyte_spider_templates_project/settings.py:9]: Ok, nice, thanks, will return them.
Update: Fixed!

File: zyte_spider_templates_project/settings.py [hunk 4]
```diff
@@ -9,25 +9,21 @@
...
 SPIDER_MIDDLEWARES = {
     "scrapy_poet.RetryMiddleware": 275,
-    "scrapy.spidermiddlewares.offsite.OffsiteMiddleware": None,
-    "zyte_spider_templates.middlewares.AllowOffsiteMiddleware": 500,
-    "zyte_spider_templates.middlewares.CrawlingLogsMiddleware": 1000,
 }
 
 # scrapy-poet
 SCRAPY_POET_DISCOVER = [
-    "zyte_spider_templates.pages",
```
Comment (kmike) [inline zyte_spider_templates_project/settings.py:30]: is it something which addon handles?

Comment (Gallaecio) [reply zyte_spider_templates_project/settings.py:30]: [It should](https://github.com/zytedata/zyte-spider-templates-private/blob/805c9a8c186943ccd8032f0e6ba431b8e97107b0/zyte_spider_templates/_addon.py#L126-L128).

Comment (PyExplorer) [reply zyte_spider_templates_project/settings.py:30]: Yes, it's there, but not sure if this aligned with this comment https://github.com/zytedata/zyte-spider-templates-project/pull/27#discussion_r1880403730

**Review by Gallaecio — APPROVED** [review]
:heavy_check_mark:, but we should probably keep this as a draft PR until we have a release of `zyte-spider-templates` with the add-on.

File: zyte_spider_templates_project/settings.py (hunk 4 above)
Comment (Gallaecio) [reply zyte_spider_templates_project/settings.py:30]: Page objects should not impact spider listing, and at run time there is no issue, so this line removal should be fine.

**Review by kmike — APPROVED** [review]

**Pull request merged**